choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerSessionDBService.java

```diff
@@ -1,237 +1,302 @@
-//          } else {
-//              throw new InvalidParameterException("Step or Session with given id(s) not found!");
-//          }
-//      } catch (final InvalidParameterException ex) {
-//          throw ex;
-//      } catch (final Exception ex) {
-//          logger.debug(ex.getMessage(), ex);
-//          throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
-//      }
-//  }
-
-//  //-------------------------------------------------------------------------------------------------
-//  @Transactional(rollbackFor = ArrowheadException.class)
-//  public ChoreographerSessionStep setRunningStepStatus(final long runningStepId, final ChoreographerStatusType status, final String message) {
-//      try {
-//          if (status == null) {
-//              throw new InvalidParameterException("Status is null or blank.");
-//          }
-//
-//          if (Utilities.isEmpty(message)) {
-//              throw new InvalidParameterException("Message is null or blank.");
-//          }
-//
-//          final Optional<ChoreographerSessionStep> runningStepOptional = choreographerRunningStepRepository.findById(runningStepId);
-//
-//          if (runningStepOptional.isPresent()) {
-//              ChoreographerSessionStep runningStepToChange = runningStepOptional.get();
-//              runningStepToChange.setStatus(status);
-//              runningStepToChange.setMessage(message);
-//              return choreographerRunningStepRepository.saveAndFlush(runningStepToChange);
-//          } else {
-//              throw new InvalidParameterException("Running step with given ID doesn't exist.");
-//          }
-//      } catch (final InvalidParameterException ex) {
-//          throw ex;
-//      } catch (final Exception ex) {
-//          logger.debug(ex.getMessage(), ex);
-//          throw new ArrowheadException(CoreCommonConstants.DATABASE_OPERATION_EXCEPTION_MSG);
-//      }
-//  }
-
-  //-------------------------------------------------------------------------------------------------
-  @Transactional(rollbackFor = ArrowheadException.class)
-  public ChoreographerSession initiateSession(final long planId, final String notifyUri) {
-      logger.debug("initiateSession started...");
+    //-------------------------------------------------------------------------------------------------
+    @Transactional(rollbackFor = ArrowheadException.class)
+    public ChoreographerSession initiateSession(final long planId, final String notifyUri) {
+        logger.debug("initiateSession started...");
+        
+        try {
+      	  final Optional<ChoreographerPlan> optional = planRepository.findById(planId);
+      	  if (optional.isEmpty()) {
+      		  worklogAndThrow("Initiating plan has been failed", new InvalidParameterException("Plan with id " + planId + " not exists"));
```

Comment:
worklog message should contains the cause too

for example, Initiating plan has been failed, because plan does not exist.